tabsie/server.R

```diff
@@ -80,17 +80,48 @@ shinyServer(
     })
 ####### BUTTON PRESSES #####################
     observeEvent(input$clearTheme, {
-      if (!valAuth) return;#break processing of not authorized.
+      #if (!valAuth) return;#break processing of not authorized.
       updateTextInput(session, "titleField", value = "")
       updateTextInput(session, "xLab", value = "")
       updateTextInput(session, "yLab", value = "")
       updateSliderInput(session, "textSize", value = 15)
       updateSliderInput(session, "xLabRotation", value = 0)
       updateSliderInput(session, "xLabHeight", value = 0)
     })
+######## DIV BOX CONTROL for IDEAS and BUGS #################
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      shinyjs::onclick("toggleIdea",toggle(id="ideaDiv",anim=T))
+      shinyjs::onclick("toggleBug",toggle(id="bugDiv",anim=T))
+    })
+
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      needIdea <- need(input$ideaInput,F);
+      needBug <- need(input$bugInput,F);
+      if(length(c(needIdea,needBug))<=1) show('emailDiv',anim=T);
+      #browser();
+      if(is.null(needIdea)) show('submitIdea') else hide('submitIdea');
+      if(is.null(needBug)) show('submitBug') else hide('submitBug');
+    })
+    
```

Comment:
Note: what actually happens is not what it looks like. This block (unfortunately) gets triggered each time the content of either field changes by even one character. Luckily the end result is what we want it to be, and it doesn't exact an unacceptable performance hit. The reason I'm saying this is, you might be tempted to "streamline" this execution by, for example, nesting the `if` statements. I tried that, and it's surprisingly hard to not cause weird and unintended behaviors. Also tried `validate()` and abandoned it for similar reasons. This works, and suggest further experimentation with conditionals be done in a new branch.